src/library_pthread.js

```diff
@@ -1043,9 +1043,13 @@ var LibraryPThread = {
     return wasDetached ? ERRNO_CODES.EINVAL : 0;
   },
 
-  // C11 threads function.
+  // C11 thread version.
   // TODO: remove this in favor or compiling musl/src/thread/pthread_detach.c
+#if USE_LSAN
+  thrd_detach: 'emscripten_builtin_pthread_detach',
+#else
   thrd_detach: 'pthread_detach',
+#endif
```

Comment:
this could be on a single line like the `{{{ }}}` stuff above.

Reply:
I couldn't make it work when the string is on the right of the ":" for some reason.  It was giving me JS compiler errors.   Plus I think this syntax is more readable in the end.

Reply:
But it works just several lines above in the diff? I don't understand. It's just adding a prefix to a string, isn't it?

Reply:
Somehow it works in the key position but not in the value position.   Oh maybe its because the keys are not in quotes..  Let me try again.